nbgrader/apps/zipcollectapp.py

```diff
@@ -0,0 +1,486 @@
+            corresponds to the `collect_step` variable in the
+            `collect_structure` config option.
+            """
+        )
+    ).tag(config=True)
+
+    extracted_directory = Unicode(
+        'extracted',
+        help=dedent(
+            """
+            The name of the directory that contains assignment submission files
+            extracted or copied from the `archive_directory`. This corresponds
+            to the `collect_step` variable in the `collect_structure` config
+            option.
+            """
+        )
+    ).tag(config=True)
+
+    timezone = Unicode(
+        "UTC",
+        help="Timezone for recording timestamps"
+    ).tag(config=True)
+
+    timestamp_format = Unicode(
+        "%Y-%m-%d %H:%M:%S %Z",
+        help="Format string for timestamps"
+    ).tag(config=True)
+
+    extractor_plugin = Type(
+        ExtractorPlugin,
+        klass=BasePlugin,
+        help=dedent(
+            """
+            The plugin class for extracting the archive files in the
+            `archive_directory`.
+            """
+        )
+    ).tag(config=True)
+
+    collector_plugin = Type(
+        FileNameCollectorPlugin,
+        klass=BasePlugin,
+        help=dedent(
+            """
+            The plugin class for processing the submitted file names after
+            they have been extracted into the `extracted_directory`.
+            """
+        )
+    ).tag(config=True)
+
+    collector_plugin_inst = Instance(FileNameCollectorPlugin).tag(config=False)
+    extractor_plugin_inst = Instance(ExtractorPlugin).tag(config=False)
+
+    @default("classes")
+    def _classes_default(self):
+        classes = super(ZipCollectApp, self)._classes_default()
+        classes.append(ExtractorPlugin)
+        classes.append(ZipCollectApp)
+        classes.append(self.collector_plugin)
+        return classes
+
+    def _format_collect_path(self, collect_step):
+        kwargs = dict(
+            downloaded=self.downloaded_directory,
+            assignment_id=self.assignment_id,
+            collect_step=collect_step,
+        )
+
+        path = os.path.join(
+            self.course_directory,
+            self.collect_directory_structure,
+        ).format(**kwargs)
+
+        return path
+
+    def _mkdirs_if_missing(self, path):
+        if not check_directory(path, write=True, execute=True):
+            self.log.warn("Directory not found. Creating: {}".format(path))
+            os.makedirs(path)
+
+    def _clear_existing_files(self, path):
+        if not os.listdir(path):
+            return
+
+        if self.force:
+            self.log.warn("Clearing existing files in {}".format(path))
+            rmtree(path)
+            os.makedirs(path)
+        else:
+            self.fail(
+                "Directory not empty: {}\nuse the --force option to clear "
+                "previously existing files".format(path)
+            )
+
+    def _find_student(self, info):
```

Comment:
I wouldn't require students to be in the database at this step -- I would allow the collector to collect any and all submisisons (which is the same behavior as `nbgrader collect`) and then have the autograder step be the one the throws an error when the student isn't in the database.